backend/grading-service/src/controllers/grading-controller.ts

```diff
@@ -18,6 +66,22 @@ const postParser = async (request: Request, response: Response) => {
 
     response.status(HttpStatusCode.OK).json({ parser });
   } catch (error) {
+    if (error instanceof ITSPostParserError) {
+      if (error.errorField === "language") {
+        response.status(HttpStatusCode.BAD_REQUEST).json({
```

Comment:
This look like DRY to me cause this piece of code appear in other place, I think you can consider exporting an method that set response status and json body from ITSPostParserError.ts and invoke it

Reply:
Oh okay, tbh the actual reason was that the ITS response on error is not very clear and consistent, so I was just using a simple way to distinguish what happens... Will polish this in the future.